packages/interchain/interchain-core/src/env.rs

```diff
@@ -138,6 +138,19 @@ pub trait InterchainEnv<Chain: IbcQueryHandler> {
     /// ```
     fn get_chain(&self, chain_id: impl ToString) -> Result<Chain, Self::Error>;
 
+    /// Returns every chain registered within the environment.
+    /// To get a single chain, use [`InterchainEnv::get_chain`]
+    /// ``` rust
+    /// use cw_orch::prelude::*;
+    /// use cw_orch_interchain::prelude::*;
+    /// use counter_contract::CounterContract;
+    /// let interchain = MockBech32InterchainEnv::new(vec![("osmosis-1","osmo"),("archway-1","arch")]);
+    ///
+    /// let all_chains = interchain.chains().unwrap();
+    ///
+    /// ```
+    fn chains(&self) -> Result<Vec<Chain>, Self::Error>;
+
```

Comment:
Can't we return iterator instead?

Reply:
And why it returns Result, at what use-case it can error?

Reply:
Yes, no Error here, I don't see why it could err. And returning iterator to chain reference as well here